wicket/src/rack_setup/config_toml.rs

```diff
@@ -0,0 +1,296 @@
+        {
+            format_multiline_array(
+                doc.get_mut(array).unwrap().as_array_mut().unwrap(),
+            );
+        }
+
+        *doc.get_mut("bootstrap_sleds").unwrap().as_array_mut().unwrap() =
+            build_sleds_array(&config.bootstrap_sleds);
+
+        populate_network_table(
+            doc.get_mut("rack_network_config").unwrap().as_table_mut().unwrap(),
+            config.rack_network_config.as_ref(),
+        );
+
+        Self { doc }
+    }
+}
+
+impl fmt::Display for TomlTemplate {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.doc.fmt(f)
+    }
+}
+
+fn format_multiline_array(array: &mut Array) {
+    for element in array.iter_mut() {
+        element.decor_mut().set_prefix(ARRAY_SEP);
+    }
+    array.set_trailing_comma(true);
+    array.set_trailing("\n");
+}
+
+fn build_sleds_array(sleds: &[BootstrapSledDescription]) -> Array {
+    let mut array = Array::new();
+    let mut prev: Option<&BootstrapSledDescription> = None;
+
+    for sled in sleds {
+        // We should never get a non-sled from wicketd; if we do, filter it out.
+        if sled.id.type_ != SpType::Sled {
+            continue;
+        }
+
+        let mut value = Formatted::new(i64::from(sled.id.slot));
+        let decor = value.decor_mut();
+
+        // We have to attach the comment for each sled on the _next_ item in the
+        // array, so here we set our prefix to be the previous item's details.
+        if let Some(prev) = prev {
+            decor.set_prefix(format!(
+                " # {serial} (model {model}, revision {rev}){sep}",
+                serial = prev.serial_number,
+                model = prev.model,
+                rev = prev.revision,
+                sep = ARRAY_SEP,
+            ));
+        } else {
+            decor.set_prefix(ARRAY_SEP);
+        }
+
+        array.push_formatted(Value::Integer(value));
+        prev = Some(sled);
+    }
+
+    // Because we attach comments to previous items, we also need to add a
+    // comment to the last element.
+    if let Some(prev) = prev {
+        array.set_trailing(format!(
+            " # {serial} (model {model}, revision {rev})\n",
+            serial = prev.serial_number,
+            model = prev.model,
+            rev = prev.revision,
+        ));
+        array.set_trailing_comma(true);
+    }
+
+    array
+}
+
+fn populate_network_table(
+    table: &mut Table,
+    config: Option<&RackNetworkConfig>,
+) {
+    // Helper function to serialize enums into their appropriate string
+    // representations.
+    fn enum_to_toml_string<T: Serialize>(value: &T) -> Cow<'static, str> {
+        let value = toml::Value::try_from(value).unwrap();
+        match value {
+            toml::Value::String(s) => Cow::from(s),
+            other => {
+                panic!("improper use of enum_to_toml_string: got {other:?}");
+            }
+        }
+    }
+
+    let Some(config) = config else {
+        return;
```

Comment:
This will happen if someone asks for the config before `wicket` has been able to get the current config from `wicketd`, which is unlikely (but not impossible) in practice. Returning the default values in the template was intentional, but assumed we fixed the default values to be reasonable (possibly just empty strings with comments explaining them).